### PR TITLE
Move all generated jobs to the new `prow-build` cluster

### PR DESCRIPTION
**Pull request by upodroid**
**Which issue(s) this PR fixes**:<br>
Move all generated jobs to the new prow-build cluster. I feel more comfortable with this change now given all the bugs we fixed when we moved autobump(#3285) and client(#3287).

If we see catastrophic failures, we can revert this PR and investigate.

/cc @dprotaso @chizhg @kvmware

### CODE REVIEW

File: prow/test.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+apiVersion: v1
```
Comment (kvmware) [inline prow/test.yaml:1]: What is this file?

Comment (upodroid) [reply prow/test.yaml:1]: random test file, shouldn't be here. Deleting

**Comment by kvmware** [comment]
/lgtm
/hold
unhold when you ready to monitor and ensure all is good

**Pull request merged**